pkg/core/native/native_nep17.go

```diff
@@ -46,8 +47,11 @@ func (c *nep17TokenNative) Metadata() *interop.ContractMD {
 }
 
 func newNEP17Native(name string, id int32) *nep17TokenNative {
-	n := &nep17TokenNative{ContractMD: *interop.NewContractMD(name, id, func(m *manifest.Manifest) {
+	n := &nep17TokenNative{ContractMD: *interop.NewContractMD(name, id, func(m *manifest.Manifest, hf config.Hardfork) {
 		m.SupportedStandards = []string{manifest.NEP17StandardName}
+		if name == nativenames.Neo && hf.Cmp(config.HFEchidna) >= 0 {
```

Comment:
NEO-specific code doesn't look nice in generic NEP-17 function, but looks like we don't have a lot of choices. Maybe except passing an additional hook into `nepNEP17Native()` (with the same `func(m *manifest.Manifest, hf config.Hardfork)` signature) and calling it if it's defined?